src/app/shared/entity/opa.ts

```diff
@@ -50,77 +67,77 @@ export class CRD {
 
 export class CRDSpec {
   names: Names;
-  validation: Validation;
+  validation?: Validation;
 }
 
 export class Names {
   kind: string;
-  shortNames: string[];
+  shortNames?: string[];
 }
 
 export class Validation {
-  openAPIV3Schema: JSONSchemaProps;
+  openAPIV3Schema?: JSONSchemaProps;
 }
 
 export class JSONSchemaProps {
```

Comment:
Are those structures used anywhere or will they be used in any near future?

Reply:
You can specify the `Validation` fields during constraint template creation, but other than that I'm not sure if there is some feature coming up (same for other structs that are related to constraints & constraint templates). 
Do you think we should only specify e.g. `object` for everything that isn't "needed" atm?

Reply:
Ye. It's hard to track then which structures are actually used by us. If it is not used in the code, I'd simply remove it and use `object`.

Reply:
okay, will change it :)

Reply:
@floreks done